### ISSUE TITLE
ERROR (MainThread) [homeassistant.components.alarm_control_panel] Platform aarlo does not generate unique IDs. ID home already exists - ignoring alarm_control_panel.aarlo_home

### ISSUE THREAD

**Issue by JeffSteinbok** [twrecked/hass-aarlo#739]
Hello,

Just upgraded to your official code vs the custom stuff I was running before, and now get this error:
ERROR (MainThread) [homeassistant.components.alarm_control_panel] Platform aarlo does not generate unique IDs. ID home already exists - ignoring alarm_control_panel.aarlo_home

In my earlier version, I think I was registering a control panel with some unique ID, not the name "home".  Wondering what's up with that.  Any thoughts?

**Comment by JeffSteinbok** [comment]
Here is what my code did:
![image](https://user-images.githubusercontent.com/7330767/235055885-57ceb15f-7085-4982-a077-230865866290.png)

**Comment by JeffSteinbok** [comment]
I haven't debugged your code, but my guess is in class ArloLocation, you're setting the unique_id to:
self._unique_id = location.entity_id
rather than 
self._unique_id = "location_" + location.id

Just a guess.

**Comment by twrecked** [comment]
I think that might be a mistake at my end. 

I was trying to tidy up the differences between the alarms and got carried away. Does patching it fix it?

**Comment by JeffSteinbok** [comment]
Patching how?

**Comment by twrecked** [comment]
Sorry, I remember what I did now, I brought the naming into line with the other devices. (You can use devices ID by setting an `aarlo` line option.)  I noticed that I had 2 home devices because Arlo gives the same name to both the user home and one shared with the user. I was working on a fix, I'll see if I can did it out.

**Comment by JeffSteinbok** [comment]
Not sure what you mean by aarlo line option. And hope it doesn’t change all the camera device names.

**Comment by JeffSteinbok** [comment]
I would have thought the internal names would be scoped to avoid collisions. That not the case?

**Comment by twrecked** [comment]
No, we don't scope the names, we've always relied on the users to name their cameras (or any device) uniquely. To prevent clashes outside of _Aarlo_ we prefix everything with `aarlo_`.  The format is general `device-type.aarlo_device_name`. (Not trying to dodge blame here) But this was how the original project worked when I forked out this version, my goal at that time was to make moving to this component as easy as possible so I didn't change it.

Some way along in the development I added the `serial_ids` option to the configuration, this will replace the `device_name` with the serial number. But as you noted, that will change all device names and probably break your current set up. But it's there if people want it.

As to why you're suddenly seeing it. The account you shared your location with has 2 homes, the one you shared with it and the one _Arlo_ gives it. I suspect they both have the same name. My earlier fix included both the shared home and account home locations - we recommend creating a second account but I have to assume at least one or two people use their main account so I have to cater for that in the code. And for you that created 2 devices with the same id.

I've pushed a fix to a branch that should fix this, I'll merge it across later.

**Comment by JeffSteinbok** [comment]
Sounds like the easy fix is for me rename some locations. :) 

Makes sense.

**Comment by twrecked** [comment]
Sorry for the confusion, and thank for the work getting this going.

Still no locations enabled in Canada, I'm relying on a VPN and US account to double check this.

**Comment by JeffSteinbok** [comment]
I'll try it out. And then rename a location to be safe.
